### ISSUE TITLE
killed at 771 iteration when training the model.

### ISSUE THREAD

**Issue by power76** [apple/turicreate#3261]
I have prepared 18 images(5 classes) and used the default background to one shot create about 27k images dataset. The synthetic process done well. But when I load the synthetic images to train the model. It always was killed at the exactly 771 iterations. 
I have a CPU-based ubuntu system(version 20) with 8G ram and 35G swap. The tensorflow version is 2.2 and batch size is 35 by default. Anybody can help me?
BTW, I have used the default backgroud and 9 images(3 classes) to train an one shot object detection model successfully.

**Comment by shantanuchhabra** [comment]
Hi @power76 
Could you share the code you're using to train the model?

Thanks.

**Comment by power76** [comment]
> Hi @power76
> Could you share the code you're using to train the model?
> 
> Thanks.

hi, the code is as below:
------------------------------------------
synthetic_data = tc.one_shot_object_detector.util.preview_synthetic_training_data(starter_images, 'label')
synthetic_data.save('training_data.sframe')
sf_loaded = tc.load_sframe('training_data.sframe')
model = tc.object_detector.create(sf_loaded, max_iterations=1500)
------------------------------------------

BTW,after one night running,the model training done and wasn't killed again. But from iteration 771 to 772, it took about 11 hours. So what's the problem?

**Comment by power76** [comment]
> Hi @power76
> Could you share the code you're using to train the model?
> 
> Thanks.

hi, All 
I noticed that I had 24726 synthetic images and the batch size set to 32. And the 24726/32=772. So after 771 iterations, it finished a full epoch. I think after one epoch there would be a large scale data reorganization, isn't it? Thanks.

**Comment by shantanuchhabra** [comment]
Thanks @power76 for the repro code! Glad that your model training completed. Before we investigate the slowness from iteration 771 to 772, could you share which version of Turi Create and which version of Python you are using?
